### PR TITLE
TBE-92 : Signed parameters for subscribe / unsubscribe

### PR DESCRIPTION
**Pull request by tofarr**
## Is PM acceptance required? (delete one)
- Yes - don't merge until JIRA issue is accepted!
## What was done?
- email_address param passed to subscribe/unsubscribe is now signed (And the signature is verified)
## How to test?
- Specs still pass
- Test on staging. Unsubscribe an email.

[TBE-92]: https://codeforamerica.atlassian.net/browse/TBE-92?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by tahsinaislam** [comment]
hi sorry this was already worked on here: https://github.com/codeforamerica/vita-min/pull/4663

File: app/controllers/notifications_settings_controller.rb [hunk 1]
```diff
@@ -31,9 +31,12 @@ def subscribe_to_emails
 
 
   def matching_intakes(email_address)
-    return if email_address.blank?
-
-    Intake.where(email_address: email_address)
+    return [] if email_address.blank?
+    email_address = Intake.signed_id_verifier.verified email_address
+    Intake.where(email_address: email_address).to_a
```
Comment (powersurge360) [inline app/controllers/notifications_settings_controller.rb:36]: Why `#to_a`?

File: app/controllers/notifications_settings_controller.rb [hunk 2]
```diff
@@ -15,7 +15,7 @@ def unsubscribe_from_emails
   def subscribe_to_emails
     matching_intakes = matching_intakes(params[:email_address])
 
-    if matching_intakes.present?
+    if matching_intakes.any?
```
Comment (powersurge360) [inline app/controllers/notifications_settings_controller.rb:18]: Probably minor, but I think if we aren't doing `#to_a` to force an array correction, you can avoid looping and do a bulk update. Maybe not even need the check at all and allow the operation to proceed on an empty collection, which might work. Not sure off the top of my head.

**Review by powersurge360 — APPROVED** [review]
Looks good. Couple minor questions that I don't mind if you don't want to go with.

**Comment by tofarr** [comment]
Closing this as TBE-92 is a duplicate of TBE-85

**Pull request closed without merging**